spring-cql/src/test/java/org/springframework/cassandra/test/integration/AbstractEmbeddedCassandraIntegrationTest.java

```diff
@@ -15,81 +15,72 @@
+ * {@link #cluster} instance to create sessions and get access. Expect the {@link #cluster} instance to be closed once
+ * the test has been run.
+ * <p>
+ * This class is intended to be subclassed by integration test classes.
+ *
  * @author Matthew T. Adams
+ * @author Mark Paluch
  */
-public class AbstractEmbeddedCassandraIntegrationTest {
-
-	public static String uuid() {
-		return UUID.randomUUID().toString();
-	}
-
-	static Logger log = LoggerFactory.getLogger(AbstractEmbeddedCassandraIntegrationTest.class);
-
-	protected static String CASSANDRA_CONFIG = "spring-cassandra.yaml";
-	protected static String CASSANDRA_HOST = "localhost";
-
-	protected static SpringCqlBuildProperties PROPS = new SpringCqlBuildProperties();
-	protected static int CASSANDRA_NATIVE_PORT = PROPS.getCassandraPort();
+public abstract class AbstractEmbeddedCassandraIntegrationTest {
 
 	/**
-	 * The session connected to the system keyspace.
+	 * Initiate a Cassandra environment in test class scope.
 	 */
-	protected static Session system;
+	@ClassRule public final static CassandraRule cassandraEnvironment = new CassandraRule("embedded-cassandra.yaml");
 
 	/**
-	 * The {@link Cluster} that's connected to Cassandra.
+	 * Initiate a Cassandra environment in test scope.
 	 */
-	protected static Cluster cluster;
-
-	public static String randomKeyspaceName() {
-		return Utils.randomKeyspaceName();
-	}
+	@Rule public final CassandraRule cassandraRule = cassandraEnvironment.testInstance()
+			.before(new SessionCallback<Object>() {
+				@Override
+				public Object doInSession(Session s) throws DataAccessException {
+					AbstractEmbeddedCassandraIntegrationTest.this.cluster = s.getCluster();
+					return null;
+				}
+			});
 
-	@BeforeClass
-	public static void startCassandra() throws TTransportException, IOException, InterruptedException,
-			ConfigurationException {
+	protected static CassandraConnectionProperties PROPS = new CassandraConnectionProperties();
 
-		EmbeddedCassandraServerHelper.startEmbeddedCassandra(CASSANDRA_CONFIG);
-	}
+	/**
+	 * The {@link Cluster} that's connected to Cassandra.
+	 */
+	protected Cluster cluster;
 
-	public static Cluster cluster() {
-		return Cluster.builder().addContactPoint(CASSANDRA_HOST).withPort(CASSANDRA_NATIVE_PORT).build();
+	public static String uuid() {
+		return UUID.randomUUID().toString();
 	}
 
 	/**
-	 * Ensures that the cluster is created and that the session {@link #SYSTEM} is connected to it.
+	 * Returns the {@link Cluster}.
+	 *
+	 * @return
 	 */
-	public static void ensureClusterConnection() {
-
-		// check cluster
-		if (cluster == null) {
-			cluster = cluster();
-		}
-
-		if (system == null) {
-			system = cluster.connect();
-		}
+	public Cluster getCluster() {
+		return cluster;
 	}
 
-	public AbstractEmbeddedCassandraIntegrationTest() {
-		ensureClusterConnection();
+	/**
+	 * Creates a new {@link Cluster}.
+	 *
+	 * @return
+	 */
+	public static Cluster createCluster() {
```

Comment:
Eventually, we can just convert all Cassandra (integration) tests over to use the `CassandraRule`, yes?  That way we won't need either the `createCluseter(..)` method or the `PROPS` (line 58).